File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> num=new HashMap<>();
        int i;
        int n=nums.length;
        int ans[]=new int[2];
        for(i=0;i<n;i++)
        {
            if(num.containsKey(target-nums[i]))
            {
                ans[0]=i;
                ans[1]=num.get(target-nums[i]);
            }
            else
            {
                num.put(nums[i],i);
            }
        }
        return ans;
    }
}

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> vec;
        unordered_map<int,int> mpp;
        for(int i=0;i<nums.size();i++){
            if(mpp.find(target-nums[i]) != mpp.end()){
                vec.emplace_back(mpp[target-nums[i]]);
                vec.emplace_back(i);
                return vec;
            }
            mpp[nums[i]]=i;
        }
        return vec;
    }
};